File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& arr) {
        int i = 0,j=0;
        while(j<arr.size()){
            if(arr[j]!=0){
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
                i++;
            }
            j++;
        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.rs
impl Solution {
    pub fn move_zeroes(arr: &mut [i32]) {
    let mut i = 0;
    let mut j = 0;
    
    while j < arr.len() {
        if arr[j] != 0 {
            if j != i {
                arr.swap(i, j);
            }
            i += 1;
        }
        j += 1;
    }
}

}